nntrainer/src/neuralnet.cpp

```diff
@@ -580,7 +601,8 @@ void NeuralNetwork::finalize() {
  */
 Tensor NeuralNetwork::forwarding(Tensor input, int &status) {
   Tensor X = input;
-  for (unsigned int i = 0; i < layers.size(); i++) {
+  /** Do not forward the loss layer, as label is not available */
+  for (unsigned int i = 0; i < layers.size() - 1; i++) {
```

Comment:
I think last layer is most likely loss layer though, it'd be better to check last layer is loss layer somewhere (maybe at `NeuralNetwork::init` or `NeuralNetwork::construct`) phase?.